CMakeLists.txt

```diff
@@ -6,8 +6,9 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
+# Disable Wredundant-decls warnings since rosidl generates redundant function declarations
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Wno-redundant-decls")
```

Comment:
We should use [add_compile_options](https://cmake.org/cmake/help/v3.0/command/add_compile_options.html), especially in modern ROS2 projects.

Reply:
Filed a new PR #60 to use `add_compile_options`. Seems `add_compile_options` can not influence the targets generated from rosidl (see [here](https://stackoverflow.com/questions/33828855/is-cmake-set-variable-recursive)), so still use `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-redundant-decls")` to suppress this warning.